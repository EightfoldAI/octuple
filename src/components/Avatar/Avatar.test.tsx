import React from 'react';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import MatchMediaMock from 'jest-matchmedia-mock';
import { Avatar, AvatarGroup } from './';

Enzyme.configure({ adapter: new Adapter() });

let matchMedia: any;

describe('Avatar', () => {
  beforeAll(() => {
    matchMedia = new MatchMediaMock();
  });

  afterEach(() => {
    matchMedia.clear();
  });

  test('Should render', () => {
    const wrapper = mount(<Avatar>AB</Avatar>);
    expect(wrapper.children().length).toEqual(1);
  });

  test('Avatar type square is default', () => {
    const wrapper = mount(<Avatar>AB</Avatar>);
    expect(wrapper.find('.avatar').hasClass('round-image')).toBeFalsy();
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Avatar type is round', () => {
    const wrapper = mount(<Avatar type={'round'}>AB</Avatar>);
    expect(wrapper.find('.round-image')).toBeTruthy();
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Avatar size is 40px', () => {
    const wrapper = mount(<Avatar size={'40px'}>AB</Avatar>);
    expect(wrapper.render()).toMatchSnapshot();
  });

  test('Should render a group', () => {
    const wrapper = mount(
      <AvatarGroup>
        <Avatar>AB</Avatar>
        <Avatar>CD</Avatar>
        <Avatar>EF</Avatar>
        <Avatar>GH</Avatar>
      </AvatarGroup>
    );
    expect(wrapper.children().length).toEqual(1);
  });

  test('Should render a list group', () => {
    const imageProps = {
      src: 'https://images.pexels.com/photos/771742/pexels-photo-771742.jpeg',
      alt: 'random profile image',
    };

    interface User {
      'data-test-id': string;
      alt: string;
      children: React.ReactNode;
      classNames: string;
      img: string;
      key: string;
      name: string;
      randomiseTheme: boolean;
    }

    const sampleList: User[] = [
      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
    ].map((i) => ({
      'data-test-id': `my-avatar-test-id-${i}`,
      alt: i === 1 ? imageProps.alt : null,
      children: `U${i}`,
      classNames: `my-avatar-class-${i}`,
      img: i === 1 ? imageProps.src : null,
      key: `key-${i}`,
      name: `User ${i}`,
      randomiseTheme: true,
    }));
    const wrapper = mount(
      <AvatarGroup
        avatarListProps={{
          items: sampleList,
          renderItem: (item: User) => (
            <Avatar
              alt={item.alt}
              classNames={item.classNames}
              data-test-id={item['data-test-id']}
              hashingFunction={() => 3}
              randomiseTheme={item.randomiseTheme}
              src={item.img}
            >
              {item.children}
            </Avatar>
          ),
        }}
      />
    );
    expect(wrapper.children().length).toEqual(1);
  });

  test('Should render a list group with max count', () => {
    const imageProps = {
      src: 'https://images.pexels.com/photos/771742/pexels-photo-771742.jpeg',
      alt: 'random profile image',
    };

    interface User {
      'data-test-id': string;
      alt: string;
      children: React.ReactNode;
      classNames: string;
      img: string;
      key: string;
      name: string;
      randomiseTheme: boolean;
    }

    const sampleList: User[] = [
      1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
    ].map((i) => ({
      'data-test-id': `my-avatar-test-id-${i}`,
      alt: i === 1 ? imageProps.alt : null,
      children: `U${i}`,
      classNames: `my-avatar-class-${i}`,
      img: i === 1 ? imageProps.src : null,
      key: `key-${i}`,
      name: `User ${i}`,
      randomiseTheme: true,
    }));
    const wrapper = mount(
      <AvatarGroup
        avatarListProps={{
          items: sampleList,
          renderItem: (item: User) => (
            <Avatar
              alt={item.alt}
              classNames={item.classNames}
              data-test-id={item['data-test-id']}
              hashingFunction={() => 3}
              randomiseTheme={item.randomiseTheme}
              src={item.img}
            >
              {item.children}
            </Avatar>
          ),
        }}
        maxProps={{
          count: 24,
        }}
      />
    );
    expect(wrapper.children().length).toEqual(1);
  });
});
