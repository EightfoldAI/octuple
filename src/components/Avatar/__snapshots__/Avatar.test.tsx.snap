// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar Avatar size is 40px 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "wrapper-style avatar avatar image-style",
      "style": "width: 40px; height: 40px; min-width: 40px; min-height: 40px; font-size: 18px;",
    },
    "children": Array [
      Node {
        "data": "AB",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar Avatar type is round 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "wrapper-style avatar avatar image-style round-image",
      "style": "width: 32px; height: 32px; min-width: 32px; min-height: 32px; font-size: 18px;",
    },
    "children": Array [
      Node {
        "data": "AB",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Avatar Avatar type square is default 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "wrapper-style avatar avatar image-style",
      "style": "width: 32px; height: 32px; min-width: 32px; min-height: 32px; font-size: 18px;",
    },
    "children": Array [
      Node {
        "data": "AB",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "text",
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "style": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
